### PR TITLE
Refactor towards supporting version upgrades

### PR DESCRIPTION
**Pull request by cevian**
Clean up the table schema to get rid of legacy tables and functionality
that makes it more difficult to provide an upgrade path.

Notable changes:
* Get rid of legacy tables and code
* Simplify directory structure for SQL code
* Simplify table hierarchy: remove root table and make chunk tables
* inherit directly from main table
* Change chunk table suffix from _data to _chunk
* Simplify schema usage: _timescaledb_internal for internal functions.
* _timescaledb_catalog for metadata tables.
* Remove postgres_fdw dependency
* Improve code comments in sql code

### CODE REVIEW

File: src/guc.c [hunk 1]
```diff
@@ -6,6 +6,8 @@
 bool		guc_disable_optimizations = false;
 bool		guc_optimize_non_hypertables = false;
 bool		guc_restoring = false;
+bool		guc_originating_node = false;
```
Comment (olofr) [inline src/guc.c]: Could `guc_originating_node` and `guc_ignore_ddl` be the same flag ? Wasn't `guc_originating_node` used to not run DDL commands ?

File: test/expected/agg_bookends.out [hunk 2]
```diff
@@ -5,7 +5,6 @@ SET client_min_messages = NOTICE;
 CREATE DATABASE single;
 \c single
 CREATE EXTENSION IF NOT EXISTS timescaledb CASCADE;
```
Comment (olofr) [inline test/expected/agg_bookends.out]: Should we drop `CASCADE` in the tests if don't have any dependencies any more ?

File: test/expected/pg_dump.out [hunk 3]
```diff
@@ -41,7 +40,7 @@ SELECT count(*)
      AND refobjid = (SELECT oid FROM pg_extension WHERE extname = 'timescaledb');
  count 
 -------
-   206
+   100
```
Comment (olofr) [inline test/expected/pg_dump.out]: 200ish -> 100, could that indicate that only one partition is dumped or is this expected ?

File: test/sql/drop_rename_hypertable.sql [hunk 4]
```diff
@@ -16,7 +16,7 @@ ALTER TABLE "renametable" RENAME TO "newname";
 SELECT * FROM "newname";
 
 SELECT * FROM _timescaledb_catalog.hypertable;
-DROP TABLE "two_Partitions";
+DROP TABLE "two_Partitions" CASCADE;
```
Comment (olofr) [inline test/sql/drop_rename_hypertable.sql]: Why is CASCADE added ?

File: scripts/docker-run.sh [hunk 5]
```diff
@@ -8,7 +8,7 @@ fi
 
 DOCKER_HOST=${DOCKER_HOST:-localhost}
 CONTAINER_NAME=${CONTAINER_NAME:-timescaledb}
-DATA_DIR=${DATA_DIR:-$PWD/data}
+DATA_DIR=${DATA_DIR-$PWD/data}
```
Comment (cevian) [inline scripts/docker-run.sh]: It needs to be - because DATA_DIR can be blank. None of the other variables can be.

Comment (RobAtticus) [inline scripts/docker-run.sh]: Ah I screwed this up then previously. Guess I thought it was a typo. Good catch.

File: scripts/docker-run.sh [hunk 6]
```diff
@@ -19,16 +19,10 @@ fi
...
   -e PGDATA=/var/lib/postgresql/data/timescaledb \
   $IMAGE_NAME $BIN_CMD \
-  -csynchronous_commit=off -cwal_writer_delay=1000 \
-  -cmax_locks_per_transaction=1000 \
-  -cshared_preload_libraries=pg_stat_statements,timescaledb \
-  -cvacuum_cost_delay=20 -cautovacuum_max_workers=1 \
-  -clog_autovacuum_min_duration=1000 -cshared_buffers=1GB \
-  -ceffective_cache_size=3GB -cwork_mem=52428kB \
-  -cmaintenance_work_mem=512MB -ctimescaledb.hostname=local \
+  -ctimescaledb.allow_install_without_preload=on \
```
Comment (cevian) [inline scripts/docker-run.sh]: Fixed (removed)

File: sql/chunk_triggers.sql [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+                                hi.main_schema_name, hi.main_index_name, hi.definition)
+        FROM _timescaledb_catalog.partition p
+        INNER JOIN _timescaledb_catalog.partition_epoch pe ON (pe.id = p.epoch_id)
+        INNER JOIN _timescaledb_catalog.hypertable_index hi ON (hi.hypertable_id = pe.hypertable_id)
+        WHERE p.id = NEW.partition_id;
+
+        --todo: can go into create_local_data_table?
+        PERFORM _timescaledb_internal.set_time_constraint(NEW.schema_name, NEW.table_name, NEW.start_time, NEW.end_time);
+        RETURN NEW;
+    ELSIF TG_OP = 'UPDATE' THEN
```
Comment (cevian) [inline sql/chunk_triggers.sql]: Fixed (removed update) will now error on update

File: src/guc.c (hunk 1 above)
Comment (cevian) [inline src/guc.c]: Good catch. Both removed (neither used).

File: test/expected/agg_bookends.out (hunk 2 above)
Comment (cevian) [inline test/expected/agg_bookends.out]: Fixed

File: test/expected/pg_dump.out (hunk 3 above)
Comment (cevian) [inline test/expected/pg_dump.out]: Means there are less objects now. This is just to make sure same as above. expected.

File: test/sql/drop_rename_hypertable.sql (hunk 4 above)
Comment (cevian) [inline test/sql/drop_rename_hypertable.sql]: Now hypertables directly inherit chunks. This is actually more correct now imho.

**Comment by RobAtticus** [comment]
Not necessary to fix in this PR, but can we agree on a standard for comments in terms of space or no space after `--`? I prefer a space just to break it up because I find it easier to read, but happy with either as long as its consistent.

**Comment by cevian** [comment]
@RobAtticus changed all -- comments to have space after `--`

**Pull request merged**